### PR TITLE
Feature/pdf visning

### PR DESCRIPTION
**Pull request by MengZhuNAV**
Preview PDF of the vedtaksbrev.
Use updated endpoint of familie-ba-sak to get the vetaksbrev.

### CODE REVIEW

File: src/frontend/komponenter/Fagsak/Vedtak/PdfFrame.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { Document, Page } from 'react-pdf/dist/entry.webpack';
+import NavFrontendSpinner from 'nav-frontend-spinner';
+
+interface IPdfFrameProps {
+    pdfData: string;
```
Comment (henninghaakonsen) [inline src/frontend/komponenter/Fagsak/Vedtak/PdfFrame.tsx:6]:
```suggestion
    file: string;
```

Maybe use the same name as the Document component.

File: src/frontend/komponenter/Fagsak/Vedtak/PdfFrame.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    const onDocumentLoadSuccess = (pdfInfo: { numPages: React.SetStateAction<number> }) => {
+        setAntallSider(pdfInfo.numPages);
+    };
+
+    return (
+        <div className="pdf-frame">
+            <Document
+                file={pdfData}
+                onLoadSuccess={onDocumentLoadSuccess}
+                error={'Kunne ikke laste inn PDF-fil.'}
```
Comment (henninghaakonsen) [inline src/frontend/komponenter/Fagsak/Vedtak/PdfFrame.tsx:21]: Use frontendFeilmelding?

File: src/frontend/typer/ressurs.ts [hunk 3]
```diff
@@ -16,6 +16,7 @@ export type Ressurs<T> =
     | {
           data: T;
           status: RessursStatus.SUKSESS;
+          melding?: string;
```
Comment (henninghaakonsen) [inline src/frontend/typer/ressurs.ts:19]: What is this used for?

File: src/frontend/komponenter/Fagsak/Vedtak/OppsummeringVedtak.tsx [hunk 4]
```diff
@@ -36,14 +37,15 @@ const OppsummeringVedtak: React.FunctionComponent<IVedtakProps> = ({ fagsak }) =
...
+                method: httpMethod,
+                url: `/familie-ba-sak/api/dokument/vedtaksbrev/${aktivtVedtak?.id}`,
             })
                 .then((response: Ressurs<string>) => {
                     if (response.status === RessursStatus.SUKSESS) {
-                        setBrev(response.data);
+                        setPdf(`data:application/pdf;base64,${response.data}`);
                         setErrorMessage(undefined);
                     } else if (response.status === RessursStatus.FEILET) {
                         setErrorMessage(response.melding);
```
Comment (henninghaakonsen) [inline src/frontend/komponenter/Fagsak/Vedtak/OppsummeringVedtak.tsx:51]: Use frontendFeilmelding? Do we need seperate error messages for pdf viewer and the rest? Why not just pass the error to PdfFrame?

Comment (MengZhuNAV) [reply src/frontend/komponenter/Fagsak/Vedtak/OppsummeringVedtak.tsx:51]: We had both HTML preview and PDF, so the error message was implemented in the upper level.
We can handle the error message in PdfFrame, but this makes PdfFrame no longer a pure viewer, and if the page gets more complicated, we still have to revert the change.
I think maybe just keep it here for now, but use the frontendFeilmelding instead. What do you think about it?

File: src/frontend/komponenter/Fagsak/Vedtak/PdfFrame.tsx (hunk 2 above)
Comment (MengZhuNAV) [reply src/frontend/komponenter/Fagsak/Vedtak/PdfFrame.tsx:21]: I think the error message here is for invalid PDF data. The backend error is displayed in the upper level component.

File: src/frontend/typer/ressurs.ts (hunk 3 above)
Comment (MengZhuNAV) [reply src/frontend/typer/ressurs.ts:19]: Dont know... I removed it.

**Pull request merged**